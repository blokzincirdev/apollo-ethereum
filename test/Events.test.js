const gql = require('graphql-tag')

const Events = artifacts.require('Events')

const config = {
  contracts: {
    Events: true,
  },
}

const QUERY_EVENTS = gql`
  query($address: Address) {
    Events(address: $address) {
      LogEvent {
        level
        message
        _timestamp
      }
    }
  }
`

contract('Events', () => {
  let client

  before(async () => {
    client = await createClient(config, { Events })
  })

  it('generates graphql schema', async () => {
    expect(client.source).to.matchSnapshot()
  })

  it('returns the correct types', async () => {
    const variables = { address: Events.address }
    const data = await client.execute(QUERY_EVENTS, variables)

    expect(data.Events.LogEvent).to.containSubset([
      { level: 'Info', message: 'Constructing contract' },
      { level: 'Warning', message: 'Warning: triggered manually' },
      { level: 'Error', message: 'Error function triggered' },
    ])

    const timestamps = data.Events.LogEvent.map(e => e._timestamp.getTime())

    expect(timestamps[0]).to.be.closeTo(Date.now(), 20000)
    expect(timestamps[1]).to.be.closeTo(Date.now(), 20000)
    expect(timestamps[2]).to.be.closeTo(Date.now(), 20000)
  })
})
