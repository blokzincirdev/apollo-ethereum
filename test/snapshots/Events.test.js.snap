// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract: Events generates graphql schema 1`] = `
"scalar Address

type Events {
  _address: Address

  \\"\\"\\"Represents a log event\\"\\"\\"
  LogEvent: [Events_LogEvent]
}

type Events_LogEvent {
  level: Events_LogLevel
  message: String
  _timestamp: Timestamp
}

enum Events_LogLevel {
  Info
  Warning
  Error
}

type EventsMutative {
  triggerWarning: Boolean
  triggerError: Boolean
}

type Mutation {
  Events(address: Address!): EventsMutative
}

type Query {
  Events(address: Address!): Events
}

scalar Timestamp
"
`;
